dotnet/src/SemanticKernel.Abstractions/Orchestration/ContextVariables.cs

```diff
@@ -120,6 +152,29 @@ public bool ContainsKey(string key)
         return this._variables.ContainsKey(key);
     }
 
+    /// <summary>
+    /// True if all the stored variables have trusted content.
+    /// </summary>
+    /// <returns></returns>
+    public bool IsAllTrusted()
+    {
+        return this._variables.Values.All(v => v.IsTrusted);
+    }
+
+    /// <summary>
+    /// Make all the variables stored in the context untrusted.
+    /// </summary>
+    public void MakeAllUntrusted()
+    {
+        // Create a copy of the variables map iterator with ToList to avoid
+        // iterating in the map while updating it
+        foreach (var item in this._variables.ToList())
+        {
+            // Note: we don't use an internal setter for better multi-threading
+            this._variables[item.Key] = new TrustAwareString(item.Value.Value, false);
```

Comment:
In this particular case, I am leaning towards keeping it immutable, what do you think? Here is my line of thinking:

The concept of trust might be relative to the context and situation, so the same string might be considered trusted in a particular scenario, but untrusted in another. So making it immutable would ensure one context will not be messing with the other (specially because there is a `.Clone()` on it) if they ever end up pointing to same object reference.

Furthermore, let's imagine we have two references to the same `TrustAwareString` being used by two different contexts and threads (A and B). In this case, if thread A modifies its reference to the `TrustAwareString` instance we might be creating a race condition with thread B, and also, if the update in place works, we might be changing the trust state or the content of a variable being used by another context and not be fully aware of it.

Regarding making it a struct, it is a really nice idea! I love it. I guess in the case, the struct would hold a reference to the encapsulated string (8 bytes in a 64 bits machine) plus a reference to the boolean (most likely 1 byte). For memory alignment optimization the compiler might decide to make it 16 bytes total anyway, but that would be really small and might fit well on the stack or inline somewhere. So the question would be how many variables we are willing to target in the `ContextVariables` map? If this number is not that big, would be nice pointing to it by value and not by refence.

I will update the code to add the missing methods that need to be implemented, such as `GetHashCode`, `Equals` and the ones in `IEquatable`, so if we decide to make it a struct, would be just a matter of updating the class definition :).

Reply:
One of the additional arguments for struct is that we can mark it as a 
`public readonly struct ...`
...which enforces the immutability!